shared/src/main/java/uk/gov/di/authentication/shared/conditions/MfaHelper.java

```diff
@@ -48,7 +56,7 @@ public static UserMfaDetail getUserMFADetail(
         var mfaMethodType = isPhoneNumberVerified ? MFAMethodType.SMS : MFAMethodType.NONE;
 
         var mfaMethod = getPrimaryMFAMethod(userCredentials);
-        if (mfaMethod.filter(MFAMethod::isMethodVerified).isPresent()) {
+        if (mfaMethod.filter(AuthAppMFAMethod::isMethodVerified).isPresent()) {
```

Comment:
Does this throw a ClassCastException if it's not an AuthAppMFAMethod?  Though I guess that's the only subclass we have at the moment.  Did MFAMethod::isMethodVerified still work as that method is on the interface?